src/Integration/Language.cs

```diff
@@ -0,0 +1,107 @@
+﻿using SonarLint.VisualStudio.Integration.Resources;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    [DebuggerDisplay("{Name} (ServerKey: {ServerKey}, ProjectType: {ProjectType}, IsSupported: {IsSupported})")]
+    public sealed class Language : IEquatable<Language>
+    {
+        public readonly static Language CSharp = new Language("cs", Strings.CSharpLanguageName, ProjectSystemHelper.CSharpProjectKind);
+        public readonly static Language VBNET = new Language("vbnet", Strings.VBNetLanguageName, ProjectSystemHelper.VbProjectKind);
+
+        public string ServerKey { get; }
+
+        public string Name { get; }
+
+        public Guid ProjectType { get; set; }
+
+        public bool IsSupported => SupportedLanguages.Contains(this);
+
+        public static IEnumerable<Language> SupportedLanguages
+        {
+            get
+            {
+                yield return CSharp;
+                // We don't support VB.NET as the corresponding VB SonarQube server plugin has been
+                // updated to support the connected experience.
+                // yield return VBNET;
+            }
+        }
+
+        public static IEnumerable<Language> KnownLanguages
+        {
+            get
+            {
+                yield return CSharp;
```

Comment:
return new []{ CSharp, VB...} is simpler and less heavy weight (there's a whole state machine created for you when you  yield return stuff).